src/P3_terminal_velocity.jl

```diff
@@ -1,141 +1,59 @@
+
 """
-    ice_particle_terminal_velocity(state, D, Chen2022, ρₐ, use_aspect_ratio)
-Returns the terminal velocity of a single ice particle as a function
-    of its size (maximum dimension, D) using the Chen 2022 parametrization.
+    ice_particle_terminal_velocity(state, velocity, ρₐ; use_aspect_ratio)
+    ice_particle_terminal_velocity(state, velocity, ρₐ, D; use_aspect_ratio)
+
+Returns the terminal velocity of a single ice particle as a function of its size 
+    (maximum dimension, `D`) using the Chen 2022 parametrization.
+
+The first method returns a function of `D`, while the second evaluates at a specific `D`.
 
 # Arguments
- - `state`: The [`P3State`](@ref) object
- - `D`: Maximum particle dimension
- - `Chen2022`: The [`CMP.Chen2022VelType`](@ref) object with terminal velocity parameters
- - `ρₐ`: Air density
+ - `state`: A [`P3State`](@ref)
+ - `velocity`: A [`CMP.Chen2022VelType`](@ref) with terminal velocity parameters
+ - `ρₐ`: Air density [kg/m³]
+ - `D`: Maximum particle dimension [m]
+
+# Keyword arguments
  - `use_aspect_ratio`: Bool flag set to `true` if we want to consider the effects
     of particle aspect ratio on its terminal velocity (default: `true`)
 """
 function ice_particle_terminal_velocity(
-    state::P3State,
-    D::FT,
-    Chen2022::CMP.Chen2022VelType,
-    ρₐ::FT,
-    use_aspect_ratio = true,
-) where {FT}
-    # TODO - tmp
-    #ρᵢ = p3_density(p3, D, F_rim, th)
-    ρᵢ = FT(916.7)
-    if D <= Chen2022.small_ice.cutoff
-        (ai, bi, ci) = CO.Chen2022_vel_coeffs_B2(Chen2022.small_ice, ρₐ, ρᵢ)
-    else
-        (ai, bi, ci) = CO.Chen2022_vel_coeffs_B4(Chen2022.large_ice, ρₐ, ρᵢ)
+    state::P3State, velocity::CMP.Chen2022VelType, ρₐ; use_aspect_ratio = true,
+)
+    function v_term(D::FT) where {FT}
+        (; small_ice, large_ice) = velocity
+        ρᵢ = FT(916.7) # ρᵢ = p3_density(p3, D, F_rim, th) # TODO: tmp
+        ϕ_factor = use_aspect_ratio ? cbrt(ϕᵢ(state, D)) : FT(1)
+        (ai, bi, ci) = if D <= small_ice.cutoff
```

Comment:
Is that better than what it was before? Or just your personal style preference?

Reply:
Let's chat about what the "what" you're referring to is here.